### PR TITLE
Tabs Functionality

### PR DESCRIPTION
**Pull request by kylealwyn**
Heyo! I just wanted to get this up here to get some review as there is a paradigm shift in how tabs are created. Would love to get some feedback and am open to any suggestions!

## Features
- Multiple tab instances
- Cmd+T to open new tab
- Cmd+O opens file in new tab

## Considerations
- Moved from Page to Tabs/Tab relationship
- App does not control creation of tabs

## Todo
- Rewrite tests to incorporate new functionality
- Tabs class needs to store activeTab when maneuvering between tabs

### CODE REVIEW

File: app/js/App.js [hunk 1]
```diff
@@ -15,13 +18,16 @@ const Page = require('./Page')
 class App {
 
   /**
-   * Start application by creating a page
+   * Start application
    */
 
   constructor(document, settings) {
+    mkdirp(path.resolve(__dirname, '..', 'tmp'))
```
Comment (wellsjo) [inline app/js/App.js]: I moved this to `index.js` on master. can you rebase?

**Comment by wellsjo** [comment]
I like what you've done here.  I have a couple comments/suggestions:
1. Maybe don't show the tabs if there is only one. It takes up a lot of screen real estate
2. Command / Control + W should close tabs, not the whole application
3. You should be able to click `x` on the active tab

**Comment by kylealwyn** [comment]
Implemented your changes and suggestions. Totally agree! Hopefully I can get to some tests tomorrow

**Comment by wellsjo** [comment]
@kylealwyn I added some comments and am thinking about a couple style changes. can you rebase with my master branch? then this is good to go

**Comment by kylealwyn** [comment]
Awesome. I can possibly get this done tomorrow morning before snowshoeing, if not then Sunday night. 

-----

Kyle Alwyn
Software Engineer | Lemonaid Health
Healthcare. Refreshingly Simple.




On Fri, Dec 16, 2016 at 3:34 PM -0800, "Wells Johnston" <notifications@github.com> wrote:












@kylealwyn I added some comments and am thinking about a couple style changes. can you rebase with my master branch? then this is good to go



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  










-- 
 

This message may contain information which is legally privileged and 
confidential. If you are not the intended recipient any unauthorized 
disclosure, copying, retention or use of this information is strictly 
prohibited.

**Comment by wellsjo** [comment]
I don't have a problem doing it and then merging, I just didn't want to mess up any work you were doing with a `git push --force`

**Comment by kylealwyn** [comment]
Rebased

**Pull request merged**